dot/core/service.go

```diff
@@ -482,49 +485,52 @@ func (s *Service) GetRuntimeVersion(bhash *common.Hash) (
 }
 
 // HandleSubmittedExtrinsic is used to send a Transaction message containing a Extrinsic @ext
-func (s *Service) HandleSubmittedExtrinsic(ext types.Extrinsic) error {
+func (s *Service) HandleSubmittedExtrinsic(ext types.Extrinsic) (isTxnValidityErr bool, err error) {
 	if s.net == nil {
-		return nil
+		return false, nil
 	}
 
 	if s.transactionState.Exists(ext) {
-		return nil
+		return false, nil
 	}
 
 	bestBlockHash := s.blockState.BestBlockHash()
 
 	stateRoot, err := s.storageState.GetStateRootFromBlock(&bestBlockHash)
 	if err != nil {
-		return fmt.Errorf("could not get state root from block %s: %w", bestBlockHash, err)
+		return false, fmt.Errorf("could not get state root from block %s: %w", bestBlockHash, err)
 	}
 
 	ts, err := s.storageState.TrieState(stateRoot)
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	rt, err := s.blockState.GetRuntime(&bestBlockHash)
 	if err != nil {
 		logger.Critical("failed to get runtime")
-		return err
+		return false, err
 	}
 
 	rt.SetContextStorage(ts)
 	// the transaction source is External
 	externalExt := types.Extrinsic(append([]byte{byte(types.TxnExternal)}, ext...))
-	txv, err := rt.ValidateTransaction(externalExt)
-	if err != nil {
-		return err
+	transactionValidity, txnValidityErr, err := rt.ValidateTransaction(externalExt)
+	if err != nil || txnValidityErr != nil {
+		if txnValidityErr != nil {
+			return true, txnValidityErr.Error()
+		}
+		return false, err
```

Comment:
It is possible to split the conditions as if a `txnValidityErr` is not nil then we return. Also, why are we returning `true, txnValidityErr.Error()` when a `txnValidityErr` is not nil?
```suggestion
if txnValidityErr != nil {
  return true, txnValidityErr.Error()
}

if err != nil {
  return false, err
}
```

Reply:
We return true in this case because in some places in the code we need to know if the error is a transactionValidityError. Before we had proper err handling we would just return a single error to represent all cases, but now that it return multiple errors we cannot do this. Since the caller function here doesn't have access to the type, I took the most straightforward approach of just returning a boolean if its a txnValidityError that the caller can act on if needed